File: src/main/java/com/hirehub/SpringServer/Entity/JobPost.java
package com.hirehub.SpringServer.Entity;

import jakarta.persistence.*;
import lombok.*;

import java.sql.Timestamp;

@Entity
@Table(name = "jobPosts")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class JobPost {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long jobId;

    @ManyToOne
    @JoinColumn(name = "companyId")
    private Company company;

    private String jobTitle;
    private String jobPosition;
    private String jobLocation;
    private String jobCategory;
    @Column(columnDefinition = "TEXT")
    private String jobDescription;

    private String requiredDegrees;
    private Integer requiredExperience;

    @Column(columnDefinition = "TEXT")
    private String requiredSkills;

    private Double minPackage;
    private Double maxPackage;
    private Integer totalOpenings;

    private String status;

    private Timestamp postedAt;
}

File: DotNetServer/Models/Entities/JobPost.cs
using System.ComponentModel.DataAnnotations;

namespace HireHubServer.Models.Entities;

public class JobPost
{
  [Key]
  public long JobId { get; set; }

  public long CompanyId { get; set; }

  public string? JobTitle { get; set; }
  public string? JobPosition { get; set; }
  public string? JobLocation { get; set; }
  public string? JobCategory { get; set; }
  public string? JobDescription { get; set; }

  public string? RequiredDegrees { get; set; }
  public int? RequiredExperience { get; set; }

  public double? MinPackage { get; set; }
  public double? MaxPackage { get; set; }
  public string? Status { get; set; }
  public DateTime PostedAt { get; set; }
}